frontend/src/metabase/actions/containers/ActionCreator/ActionCreator.tsx

```diff
@@ -94,33 +98,37 @@ function ActionCreator({
 
   const [showSaveModal, setShowSaveModal] = useState(false);
 
-  const isEditable = model.canWriteActions();
+  const isEditable = isNew || model.canWriteActions();
 
   const handleCreate = async (values: CreateActionFormValues) => {
     if (action.type !== "query") {
       return; // only query action creation is supported now
     }
 
-    await onCreateAction({
+    const reduxAction = await onCreateAction({
       ...action,
       ...values,
       visualization_settings: formSettings,
     } as WritebackQueryAction);
+    const createdAction = Actions.HACK_getObjectFromAction(reduxAction);
 
     // Sync the editor state with data from save modal form
     handleActionChange(values);
 
     setShowSaveModal(false);
+    onSubmit?.(createdAction);
     onClose?.();
   };
 
-  const handleUpdate = () => {
+  const handleUpdate = async () => {
     if (isSavedAction(action)) {
-      onUpdateAction({
+      const reduxAction = await onUpdateAction({
         ...action,
         model_id: model.id(),
         visualization_settings: formSettings,
       });
+      const updatedAction = Actions.HACK_getObjectFromAction(reduxAction);
```

Comment:
It seems that we only need the returned `action` to call `handleActionCreated` which changes the url. Can we extract everything to a single redux action instead of dealing with redux here?

Reply:
@ranquild, sorry, I'm not sure what you mean here. Do you suggest adding another redux action that'd run `Actions.actions.update` and then return the new writeback actions via `HACK_getObjectFromAction`?

Reply:
I'm not sure it's going to play well with other occurrences of the action editor. We also use the editor in dashboards, so you can create a new action right from there instead of getting to the model detail. We won't need to navigate to the model detail in that case

Reply:
FWIW, I think it'd make sense to move these from `ActionCreator` to corresponding contexts together with [this](https://github.com/metabase/metabase/pull/28327#discussion_r1109122672)